packages/third-parties/src/google/ga.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+// TODO: Evaluate import 'client only'
+import React, { useEffect } from 'react'
+import Script from 'next/script'
+
+declare global {
+  interface Window {
+    dataLayer?: Object[]
+  }
+}
+
+type GAParams = {
+  gaId: string
+}
+
+export function GoogleAnalytics(props: GAParams) {
+  const { gaId } = props
+
+  useEffect(() => {
+    // performance.mark is being used as a feature use signal. While it is traditionally used for performance
+    // benchmarking it is low overhead and thus considered safe to use in production and it is a widely available
+    // existing API.
+
+    performance.mark('mark_feature_usage', {
```

Comment:
Hey, you can find more details in this PR - https://github.com/vercel/next.js/pull/57439

Basically, this will be measured by the Chrome Speed Metrics team's API. They will consume any `performance.mark` with this signature as a feature we want to track.

Reply:
Gotcha, might be better if we can update the comments to sth that the performance measurement will be done on google side, so that we won't get confused when we didn't see any `performance.measure` call in the code, and would think they're unused metrics

Reply:
Done